src/Terrain/MapWindow.java

```diff
@@ -71,7 +71,7 @@ public Environment getEnvironment(int w, int h)
   /** Returns the Location at the specified location, or null if that Location does not exist. */
   public Location getLocation(XYCoord location)
   {
-    return getLocation(location.xCoord, location.yCoord);
+    return master.getLocation(location);
```

Comment:
Why change this to call master directly? Now the behavior is different depending on whether I call `getLocation(XYCoord)` vs `getLocation(int, int)`. I think this enables anybody to see truth data about what's behind fog, too.

Reply:
You're completely right, I'm blind.
I wanted to be lazy/DRY, but I missed the oncoming headlights.